v_5_2_0/files/conf/k8s-addons

```diff
@@ -88,29 +88,29 @@ $KUBECTL delete pods -l k8s-app=kube-proxy -n kube-system
 
 {{ if not .DisableCalico -}}
 
-# apply calico
-CALICO_FILE="calico-all.yaml"
+{{ if eq .Cluster.Kubernetes.CloudProvider "aws" -}}
+## Apply AWS VPC CNI and Calico for ensuring network policies (only on AWS)
+CNI_FILE="aws-cni.yaml calico-policy-only.yaml"
+{{ else -}}
+## Apply Calico with all its components
+CNI_FILE="calico-all.yaml"
+{{ end -}}
 
 while
-    $KUBECTL apply -f /srv/$CALICO_FILE
+    $KUBECTL apply -f /srv/$CNI_FILE
     [ "$?" -ne "0" ]
 do
     echo "failed to apply /srv/$manifest, retrying in 5 sec"
     sleep 5s
 done
 
-# wait for healthy calico - we check for pods - desired vs ready
-while
-    # result of this is 'eval [ "$DESIRED_POD_COUNT" -eq "$READY_POD_COUNT" ]'
-    $KUBECTL -n kube-system  get ds calico-node 2>/dev/null >/dev/null
-    RET_CODE_1=$?
-    eval $($KUBECTL -n kube-system get ds calico-node | tail -1 | awk '{print "[ \"" $2"\" -eq \""$4"\" ] "}')
-    RET_CODE_2=$?
-    [ "$RET_CODE_1" -ne "0" ] || [ "$RET_CODE_2" -ne "0" ]
-do
-    echo "Waiting for calico to be ready . . "
-    sleep 3s
-done
+{{ if eq .Cluster.Kubernetes.CloudProvider "aws" -}}
+echo "Waiting for aws-node to be ready..."
+$KUBECTL -n kube-system -l k8s-app=aws-node wait --for=condition=Ready --timeout=10m pods
+{{ end -}}
+
+echo "Waiting for calico-node to be ready..."
+$KUBECTL -n kube-system -l k8s-app=calico-node wait --for=condition=Ready --timeout=10m pods
```

Comment:
This is so much better.

Reply:
Actually I have spent some time to refactor the entire k8s-adodns file using helpers and only kubectl and it looks much leaner. but never had the chance to commit it.